### PR TITLE
Add single select attribute filter plugin

### PR DESCRIPTION
**Pull request by kandl**
- The plugin is able to render attribute filters with specified display forms as a single select attribute filters.

JIRA: RAIL-4546

### CODE REVIEW

**Review by ivanmjartan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-plugin-examples-gate-PR25

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-plugin-examples-fossa-scanning-zuul-el7/23/ : FAILURE in 2m 18s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-plugin-examples-sonarqube-scan-zuul-gate/23/ : SUCCESS in 2m 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-plugin-examples-merge-compliance-check/24/ : SUCCESS in 36s

File: single_select_plugin/package.json [hunk 1]
```diff
@@ -0,0 +1,120 @@
+{
+    "name": "single-select-plugin",
+    "version": "1.0.0",
+    "description": "",
+    "author": "",
+    "private": true,
+    "license": "MIT",
```
Comment (andriichumak) [inline single_select_plugin/package.json:7]: Consider linking to the root license file instead, e.g.: `SEE LICENSE IN ../LICENSE`
See how it's done in other plugins: https://github.com/gooddata/gooddata-plugin-examples/blob/master/dashboard_description_plugin/package.json#L7

File: single_select_plugin/src/dp_single_select_plugin/Plugin.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import entryPoint from "../dp_single_select_plugin_entry";
+
+export class Plugin extends DashboardPluginV1 {
+    public readonly author = entryPoint.author;
+    public readonly displayName = entryPoint.displayName;
+    public readonly version = entryPoint.version;
+    public readonly minEngineVersion = entryPoint.minEngineVersion;
+    public readonly maxEngineVersion = entryPoint.maxEngineVersion;
+
+    public onPluginLoaded(_ctx: DashboardContext, _parameters?: string): Promise<void> | void {
```
Comment (andriichumak) [inline single_select_plugin/src/dp_single_select_plugin/Plugin.tsx:14]: I would remove the methods that are not used. I.e. `onPluginLoaded` and `onPluginUnload`.

File: single_select_plugin/src/harness/constants.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// (C) 2021-2022 GoodData Corporation
+export const DEFAULT_WORKSPACE = "xms7ga4tf3g3nzucd8380o2bev8oeknp";
```
Comment (andriichumak) [inline single_select_plugin/src/harness/constants.ts:2]: Do we want to expose this IDs?

It's hidden in other plugins: https://github.com/gooddata/gooddata-plugin-examples/blob/master/dashboard_description_plugin/src/harness/constants.ts

**Review by andriichumak — COMMENTED** [review]
Thanks, looks great! See a few minor comments.

**Pull request merged**